### DISCUSSION TITLE
Proposal to Support Swift Package Manager in PeripheryKit to Enhance Collaboration and Maintainability

### DISCUSSION THREAD

**Discussion by rofle100lvl** [peripheryapp/periphery#749] [General]
I am currently developing an open-source tool aimed at analyzing and visualizing project dependency graphs. During this process, I've come to realize that my project shares similar objectives with PeripheryKit. Inspired by the potential synergy, I took a closer look at your codebase and appreciated the clean and structured implementation you have maintained.

To leverage PeripheryKit more effectively in my project and potentially aid others, I propose introducing Swift Package Manager (SPM) support for PeripheryKit. This addition would significantly streamline the integration process for numerous developers who rely on SPM for their dependency management, including myself.

Incorporating SPM support would not only broaden the reach and usability of PeripheryKit but also attract new contributors who can further enhance its capabilities. Additionally, maintaining SPM compatibility would encourage us to uphold the high standards of code quality and organization that PeripheryKit exemplifies.

Looking forward to your thoughts.

**Comment by ileitch** [comment]
This sounds reasonable to me, I'm happy to enable developers to use whatever parts of Periphery they need. I think in the future we should think about what a proper API looks like, but for now exposing PeripheryKit works.
